public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -27,66 +26,75 @@ const SubmissionsTable = () => {
         return assignments
     }
 
-    const NewSubmissionsTable = () => {
-        const table: React.JSX.Element[] = []
-        sortedAssignments().forEach(assignment => {
-            const courseID = assignment.CourseID
-            const submissions = state.submissions.ForAssignment(assignment)
-            if (!submissions) {
+    const table: React.JSX.Element[] = []
+    sortedAssignments().forEach(assignment => {
+        const courseID = assignment.CourseID
+        const submissions = state.submissions.ForAssignment(assignment)
+        if (!submissions) {
+            return
+        }
+        if (state.enrollmentsByCourseID[courseID.toString()]?.status !== Enrollment_UserStatus.STUDENT) {
+            return
+        }
+        // Submissions are indexed by the assignment order - 1.
+        const submission = submissions.find(sub => sub.AssignmentID === assignment.ID) ?? create(SubmissionSchema)
+        const status = getStatusByUser(submission, state.self.ID)
+        if (!isApproved(status) && assignment.deadline) {
```

Comment:
Don't mix logic related to the deadline guard check and approved status. I think it will make more sense to do the deadline guard separately first (could be done on Line 31 before anything else in the `forEach` loop):
```tsx
if (isExpired(assignment)) {
    return  // ignore expired assignments
}
```

```tsx
// isExpired returns true if the assignment's deadline has expired.
// An assignment is considered expired if its deadline expired more than one month ago.
function isExpired(assignment) {
    const date = timestampDate(assignment.deadline)
    const now = new Date()
    return (
        date.getFullYear() !== now.getFullYear() ||
        date.getMonth() > now.getMonth() + 1
    )
}
```